process/block/metablock.go

```diff
@@ -277,20 +277,33 @@ func (mp *metaProcessor) RestoreBlockIntoPools(headerHandler data.HeaderHandler,
 		hdrHashes = append(hdrHashes, shardData.HeaderHash)
 	}
 
-	hdrsBuff, err := mp.store.GetAll(dataRetriever.BlockHeaderUnit, hdrHashes)
-	if err != nil {
-		return err
-	}
+	for _, hdrHash := range hdrHashes {
+		buff, err := mp.store.Get(dataRetriever.BlockHeaderUnit, hdrHash)
+		if err != nil {
+			log.Error(err.Error())
+			continue
+		}
 
-	for hdrHash, hdrBuff := range hdrsBuff {
 		hdr := block.Header{}
-		err = mp.marshalizer.Unmarshal(&hdr, hdrBuff)
+		err = mp.marshalizer.Unmarshal(&hdr, buff)
 		if err != nil {
-			return err
+			log.Error(err.Error())
+			continue
 		}
 
 		headerPool.Put([]byte(hdrHash), &hdr)
 
+		err = mp.store.GetStorer(dataRetriever.BlockHeaderUnit).Remove([]byte(hdrHash))
+		if err != nil {
+			log.Error(err.Error())
+		}
+
+		nonceToByteSlice := mp.uint64Converter.ToByteSlice(hdr.Nonce)
+		err = mp.store.GetStorer(dataRetriever.ShardHdrNonceHashDataUnit).Remove(nonceToByteSlice)
```

Comment:
put it into the shardhdrnonces pool as well.

Reply:
Done